reconcile/prometheus_rules_tester.py

```diff
@@ -0,0 +1,269 @@
+                                 'check_result': 'Cannot parse test'})
+            continue
+
+        # this makes things so much simpler. Let's revisit it if we need
+        if len(rule_files) > 1:
+            failed_tests.append({**test_to_run,
+                                 'check_result': 'Only 1 rule file per test'})
+            continue
+
+        if rule_files[0] not in rules:
+            msg = f'rule file {rule_files[0]} does not exist'
+            failed_tests.append({**test_to_run, 'check_result': msg})
+            continue
+
+        try:
+            for cluster, namespaces in rules[rule_files[0]].items():
+                for namespace, rule_data in namespaces.items():
+                    variables = rule_data.get('variables', {})
+                    test_yaml_spec = \
+                        yaml.safe_load(
+                            orb.process_extracurlyjinja2_template(
+                                body=test,
+                                vars=variables))
+                    test_yaml_spec.pop('$schema')
+
+                    tests_to_run.append({
+                        **test_to_run,
+                        'test': test_yaml_spec,
+                        'rule_files': {rule_files[0]: rule_data['spec']},
+                        'namespace': namespace,
+                        'cluster': cluster
+                    })
+        except Exception as e:
+            logging.warning(traceback.format_exc())
+            msg = f'Error in test template for cluster {cluster} and ' \
+                  f'namespace {namespace}: {e}'
+            failed_tests.append({**test_to_run, 'check_result': msg})
+            continue
+
+    result = threaded.run(func=run_test,
+                          iterable=tests_to_run,
+                          thread_pool_size=thread_pool_size)
+
+    failed_tests.extend([test for test in result if not test['check_result']])
+
+    return failed_tests
+
+
+def run(dry_run, thread_pool_size=10, cluster_name=None):
+    orb.QONTRACT_INTEGRATION = QONTRACT_INTEGRATION
+    orb.QONTRACT_INTEGRATION_VERSION = QONTRACT_INTEGRATION_VERSION
+
+    rules = get_prometheus_rules(cluster_name)
+    invalid_rules = check_prometheus_rules(rules, thread_pool_size)
+    if invalid_rules:
+        for i in invalid_rules:
+            logging.warning(f"Error in rule {i['path']} from namespace "
+                            f"{i['namespace']} in cluster "
+                            f"{i['cluster']}:  {i['check_result']}")
+
+    tests = get_prometheus_tests()
+    failed_tests = check_prometheus_tests(tests, rules, thread_pool_size)
+    if failed_tests:
+        for f in failed_tests:
+            msg = f"Error in test {f['path']}"
+            if 'cluster' in f:
+                msg += f" from namespace {f['namespace']} in cluster" \
+                       f"{f['cluster']}"
+            msg += f":  {f['check_result']}"
+
+            logging.warning(msg)
```

Comment:
if this eventually leads to exit with error, should we log error?

Reply:
Yes, it makes sense.

Reply:
fixed in 248412f